### PR TITLE
[FLINK-20879][table-api] Use MemorySize type instead of String type for memory ConfigOption in ExecutionConfigOptions

### PR DESCRIPTION
**Pull request by xiaoHoly**
## What is the purpose of the change

Use MemorySize type instead of String type for memory ConfigOption in ExecutionConfigOptions

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not documented)

### CODE REVIEW

**Comment by xiaoHoly** [comment]
cc,@wuchong .please review this change,thanks

**Comment by xiaoHoly** [comment]
cc,@godfreyhe

**Comment by wuchong** [comment]
Hi @xiaoHoly , could you rebase the branch to resovle the conflicts?

**Comment by xiaoHoly** [comment]
@wuchong ,Jark, I hava resovle the conflicts.

**Comment by wuchong** [comment]
Thanks @xiaoHoly , I will help to review this.

File: flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/plan/nodes/physical/batch/BatchPhysicalSortMergeJoin.scala [hunk 1]
```diff
@@ -189,7 +189,7 @@ class BatchPhysicalSortMergeJoin(
       rightAllKey,
       filterNulls,
       condition,
-      estimateOutputSize(getLeft) < estimateOutputSize(getRight),
+     estimateOutputSize(getLeft) < estimateOutputSize(getRight),
```
Comment (wuchong) [inline flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/plan/nodes/physical/batch/BatchPhysicalSortMergeJoin.scala:192]: Please revert this.

File: flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/utils/ExecNodeUtil.java [hunk 2]
```diff
@@ -42,8 +42,8 @@
      *
      * <p>TODO: This method can be removed once FLINK-20879 is finished.
      */
-    public static long getMemorySize(TableConfig tableConfig, ConfigOption<String> option) {
-        return MemorySize.parse(tableConfig.getConfiguration().getString(option)).getBytes();
+    public static long getMemorySize(TableConfig tableConfig, ConfigOption<MemorySize> option) {
```
Comment (wuchong) [inline flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/utils/ExecNodeUtil.java:45]: I think we don't need this utility method anymore.

Comment (xiaoHoly) [reply flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/utils/ExecNodeUtil.java:45]: okey~

File: flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/utils/ExecNodeUtil.java [hunk 3]
```diff
@@ -37,15 +37,6 @@
 
 /** An Utility class that helps translating {@link ExecNode} to {@link Transformation}. */
 public class ExecNodeUtil {
-    /**
```
Comment (wuchong) [inline flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/utils/ExecNodeUtil.java:40]: Remember to remove useless imports.

**Review by wuchong — CHANGES_REQUESTED** [review]
Please check the build log, there is still other format problems.

**Comment by wuchong** [comment]
You can setup the formatting plugin which can help you auto-format the code. 
https://ci.apache.org/projects/flink/flink-docs-master/flinkDev/ide_setup.html#code-formatting

**Comment by xiaoHoly** [comment]
@wuchong ,Thanks for  "setup"  doc

**Comment by wuchong** [comment]
The build is still failed.

**Comment by wuchong** [comment]
The build is still failed.

**Comment by wuchong** [comment]
The following case is failed:

```
[ERROR] Tests run: 11, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 13.404 s <<< FAILURE! - in org.apache.flink.table.client.gateway.local.ExecutionContextTest
[ERROR] testConfiguration(org.apache.flink.table.client.gateway.local.ExecutionContextTest)  Time elapsed: 0.018 s  <<< FAILURE!
java.lang.AssertionError: expected:<128kb> but was:<128 kb>
```

**Comment by xiaoHoly** [comment]
> The following case is failed:
> 
> ```
> [ERROR] Tests run: 11, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 13.404 s <<< FAILURE! - in org.apache.flink.table.client.gateway.local.ExecutionContextTest
> [ERROR] testConfiguration(org.apache.flink.table.client.gateway.local.ExecutionContextTest)  Time elapsed: 0.018 s  <<< FAILURE!
> java.lang.AssertionError: expected:<128kb> but was:<128 kb>
> ```

done

**Review by wuchong — APPROVED** [review]
LGTM.

**Comment by xiaoHoly** [comment]
Thanks for review, @wuchong

**Comment by wuchong** [comment]
@xiaoHoly , there is a conflict, could you rebase the branch again?

**Comment by xiaoHoly** [comment]
ok,a piece of cake

**Comment by xiaoHoly** [comment]
The conflict has been resolved @wuchong

**Comment by wuchong** [comment]
The compile is failed again...

**Comment by xiaoHoly** [comment]
> The compile is failed again...

fix again

**Comment by xiaoHoly** [comment]
Hi,@wuchong ,Test - libraries error,How to fix?

**Comment by wuchong** [comment]
We may need to retrigger the build.

**Comment by wuchong** [comment]
@flinkbot run azure

**Comment by wuchong** [comment]
The docs tests are failed: https://dev.azure.com/apache-flink/apache-flink/_build/results?buildId=12037&view=logs&j=fc5181b0-e452-5c8f-68de-1097947f6483&t=62110053-334f-5295-a0ab-80dd7e2babbf&l=33256

```
[INFO] 
[INFO] Results:
[INFO] 
[ERROR] Failures: 
[ERROR]   ConfigOptionsDocsCompletenessITCase.testCompleteness:77->compareDocumentedAndExistingOptions:221 Documentation is outdated, please regenerate it according to the instructions in flink-docs/README.md.
	Problems:
		Documentation of table.exec.spill-compression.block-size in ExecutionConfigOptions is outdated. Expected: default=(64 kb) description=(The memory size used to do compress when spilling data. The larger the memory, the higher the compression ratio, but more memory resource will be consumed by the job.).
		Documented option table.exec.spill-compression.block-size does not exist.
[INFO] 
[ERROR] Tests run: 1, Failures: 1, Errors: 0, Skipped: 0
[INFO] 
[INFO] ------------------------------------------------------------------------
```

**Comment by xiaoHoly** [comment]
It looks like the build  was successful

**Pull request merged**